server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1071,6 +1090,16 @@ private DefaultSearchContext createSearchContext(ReaderContext reader, ShardSear
         return searchContext;
     }
 
+    static boolean concurrentSearchEnabled(ResultsType resultsType, SearchSourceBuilder source) {
+        if (resultsType != ResultsType.DFS) {
+            return false; // only enable concurrent collection for DFS phase for now
+        }
+        if (source != null && source.aggregations() != null) {
```

Comment:
shall we look at aggs only when results type is query? Being paranoid :)

Reply:
So far we would return in the condition above. I think this method needs adaption anyway when that part changed and we allow the other result types.